src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -117,16 +125,17 @@ public String getScript() {
             }
             scriptEncodingCharset = zOSSystemEncodingCharset.name();
         }
-        
+
         ShellController c = new ShellController(ws,(os == OsType.ZOS));
         FilePath shf = c.getScriptFile(ws);
 
         shf.write(script, scriptEncodingCharset);
 
         final Jenkins jenkins = Jenkins.getInstance();
+        String shell = null;
         String interpreter = "";
```

Comment:
nit: Maybe a bit clearer to remove the `interpreter` variable here and instead pass `shell` to `scriptLauncherCmd` and compute `interpreter` inside of that method, since that is what `binaryLauncherCmd` does. (Just thinking of how to make the code path as similar as possible for both cases.)

Reply:
Done in ef70f18.